scene/main/viewport.cpp

```diff
@@ -1691,8 +1691,26 @@ Control *Viewport::_gui_find_control_at_pos(CanvasItem *p_node, const Point2 &p_
 	}
 
 	Control *c = Object::cast_to<Control>(p_node);
+	Point2 point = matrix.affine_inverse().xform(p_global);
+
+	if (!c || !c->is_clipping_contents() || c->has_point(point)) {
+		// If node is a Container, it can optimize the search at the given position.
+		Container *cnt = Object::cast_to<Container>(p_node);
```

Comment:
```suggestion
		Container *cont = Object::cast_to<Container>(p_node);
```
Let's not call it that lol 🙃

Reply:
Yeah, good catch :sweat_smile: